ocean/helium/data_transfer.py

```diff
@@ -123,17 +137,29 @@ def _download_dir(bucket, prefix, dest_path):
     with tqdm(total=total_size, unit='B', unit_scale=True) as progress:
         callback = ProgressCallback(progress)
 
+        metadata = {}
+        lock = Lock()
+
         futures = []
         for key, dest_file, size in tuples_list:
+            def meta_callback(resp):
+                meta = _parse_metadata(resp)
+                with lock:
+                    metadata[key] = meta
```

Comment:
Why do you need to lock here? Are the keys not unique in tuple_list?

Reply:
They are... But dicts aren't thread-safe?

Reply:
I thought all the built in types with threadsafe, but according to the google style guide I guess we should still lock. Thanks for clarifying.

Do not rely on the atomicity of built-in types.

While Python’s built-in data types such as dictionaries appear to have atomic operations, there are corner cases where they aren’t atomic (e.g. if hash or eq are implemented as Python methods) and their atomicity should not be relied upon. Neither should you rely on atomic variable assignment (since this in turn depends on dictionaries).